crates/eth-rpc/tests/kkrt_prep.rs

```diff
@@ -0,0 +1,377 @@
+        let contract_class = Arc::new(legacy_contract);
+
+        let res = account.declare_legacy(contract_class).send().await.unwrap();
+        // TODO add assert
+        let _receipt = account.provider().get_transaction_receipt(res.transaction_hash).await.unwrap();
+
+        let filename =
+            path.file_stem().expect("File has no stem").to_str().expect("Cannot convert filename to string").to_owned();
+
+        class_hash.insert(filename, res.class_hash);
+    }
+    class_hash
+}
+
+async fn deploy_and_fund_eoa(
+    account: &SingleOwnerAccount<JsonRpcClient<HttpTransport>, LocalWallet>,
+    kkrt_address: FieldElement,
+    amount: FieldElement,
+    eoa_account_address: FieldElement,
+    fee_token_address: FieldElement,
+) -> FieldElement {
+    let call_get_starknet_address = FunctionCall {
+        contract_address: kkrt_address,
+        entry_point_selector: get_selector_from_name("compute_starknet_address").unwrap(),
+        calldata: vec![eoa_account_address],
+    };
+
+    let eoa_account_starknet_address_result =
+        account.provider().call(call_get_starknet_address, BlockId::Tag(BlockTag::Latest)).await;
+
+    let deployment_of_eoa_account_result = account
+        .execute(vec![Call {
+            calldata: vec![eoa_account_address],
+            // devnet UDC address
+            to: kkrt_address,
+            selector: get_selector_from_name("deploy_externally_owned_account").unwrap(),
+        }])
+        .send()
+        .await
+        .unwrap();
+
+    // TODO: add check here
+    let _deployment_of_eoa_account_result_receipt =
+        account.provider().get_transaction_receipt(deployment_of_eoa_account_result.transaction_hash).await.unwrap();
+
+    let eoa_account_starknet_address = *eoa_account_starknet_address_result.unwrap().first().unwrap();
+
+    let amount_high = FieldElement::from_dec_str("0").unwrap();
+    let transfer_calldata = vec![eoa_account_starknet_address, amount, amount_high];
+
+    let transfer_res = account
+        .execute(vec![Call {
+            calldata: transfer_calldata,
+            // eth fee addr
+            to: fee_token_address,
+            selector: get_selector_from_name("transfer").unwrap(),
+        }])
+        .send()
+        .await
+        .unwrap();
+
+    // TODO: add checks like the python deploy script has
+    let _transfer_receipt = account.provider().get_transaction_receipt(transfer_res.transaction_hash).await.unwrap();
+
+    let call_get_balance_of_starknet_address_of_eoa = FunctionCall {
+        contract_address: fee_token_address,
+        entry_point_selector: get_selector_from_name("balanceOf").unwrap(),
+        calldata: vec![eoa_account_starknet_address],
+    };
+
+    let _balance_of_starknet_address_of_eoa_result =
+        account.provider().call(call_get_balance_of_starknet_address_of_eoa, BlockId::Tag(BlockTag::Latest)).await;
+
+    let call_get_nonce_of_starknet_address_of_eoa = FunctionCall {
```

Comment:
I think we don't need to do this, we can simply get the nonce from the starknet client itself -> we should deprecate this nonce function soon